apyros/metalog.py

```diff
@@ -51,12 +51,14 @@ def getLog( self, prefix ):
             self.f.flush()
             return filename
 
-        for line in self.f:
+        for i, line in enumerate(self.lines):
             print "LINE", line.strip()
             if line.startswith( prefix + ':' ):
                 ret = line.split()[1].strip()
                 assert ret.startswith("logs/")
+                self.lines = self.lines[i+1:]  # cut already passed lines
```

Comment:
If matching line is found then drop all previous lines. If no matching is found keep original list (as that particular source was probably not logged).